### ISSUE TITLE
Add x as gate bypass target (fixes eml(x,x) recovery)

### ISSUE THREAD

**Issue by oaustegard** [oaustegard/eml-sr#4]
**Bug:** The SR engine cannot discover `f(x) = exp(x) - ln(x)` even though this is just `eml(x, x)` — the simplest possible depth-1 expression with both inputs equal to x.

**Root cause:** Each internal node's gate can only bypass its child input to the constant `1`:
```python
left_in = s * 1 + (1-s) * child   # s=1 → 1, s=0 → child
```
There's no way to route `x` through the gate — only through leaves.

**Fix options:**

**Option A (minimal):** Change gate from binary sigmoid to 3-way softmax over `[1, x, child]`:
```python
# Current: sigmoid → blend between 1 and child
# Proposed: softmax → blend between 1, x, and child
gate_probs = softmax(gate_logits / tau, dim=-1)  # (n_internal, 2, 3)
input = gate_probs[..., 0] * 1 + gate_probs[..., 1] * x + gate_probs[..., 2] * child
```
This adds `n_internal * 2` parameters (one extra logit per gate per side).

**Option B (paper's approach):** Use the `α + βx + γf` parameterization from Section 4.3, equation (6). Each input is a learned linear combination of `1`, `x`, and the child output `f`. This is more general and matches the paper exactly.

**Impact:** Option B subsumes Option A and matches the paper. Requires changing leaf logits from `(n_leaves, 2)` to `(n_leaves, 2)` (unchanged — leaves don't have children) and gate logits from `(n_internal, 2)` to `(n_internal, 2, 3)` (add x routing).

**Test case:** `eml(x, x)` = `exp(x) - ln(x)` should be recoverable at depth 1.
